src/analysis/rule_extraction.jl

```diff
@@ -3,26 +3,26 @@ using StableRNGs: StableRNG
 import ..performance: temporal_variability
 
 """
-    Rule{V<:Vector{Float64},W<:Vector{Vector{Any}}}
+    Rule{V<:Vector{Vector},W<:Vector{Float64}}
 
-A Rule contains a condition (vector of conditional clauses) and consequent (vector 
+A Rule contains a condition (vector of conditional clauses) and consequent (vector
 of values for then and else results of the condition)
 """
-struct Rule{V<:Vector{Float64},W<:Vector{Vector{Any}}}
-    condition::W
-    consequent::V
+struct Rule{V<:Vector{Vector},W<:Vector{Float64}}
+    condition::V
+    consequent::W
 end
 
 """
-    rules(rules::SIRUS.StableRules)
+    rules(rules::SIRUS.StableRules{Float64})
 
 Collates vector of Rule objects. These are **not** the same as SIRUS.Rule object.
 
 # Arguments
 - `rules` : SIRUS.StableRules object
 
-# Return
-Vector{ADRIA.analysis.Rule{Vector{Float64}, Vector{Vector{Any}}}}
+# Returns
+Vector{ADRIA.analysis.Rule{Vector{Float64}, Vector{Vector}}}
 """
 function rules(rules::SIRUS.StableRules{Float64})
```

Comment:
Docstring should detail purpose more so than the technical "what".

Suggest adding a return type to function signature:

```julia
function rules(rules::SIRUS.StableRules{Float64})::Vector{Rule{Vector{Float64}, Vector{Vector}}}
```

Or at least:

```julia
function rules(rules::SIRUS.StableRules{Float64})::Vector
```

Reply:
I am adding more details about the purpose/meaning of the Rules object, but I have one question: this is a Vector of Rules, so my idea is to give more details at the Rules object Docstring and leave this one as it is. Do you think this make sense or do you think there is something specific I should add to this one?

Reply:
You can add a  `# See also` section to the docstring and point the reader to the documentation for `Rules`